src/transformStateWithClones.js

```diff
@@ -8,6 +8,36 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let stateClone = { ...state };
+  const array = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        stateClone = {};
+        break;
+
+      case 'addProperties':
+        stateClone = { ...stateClone, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        stateClone = { ...stateClone };
```

Comment:
There's no need to create a new copy of `stateClone` with `{ ...stateClone }` before the `forEach` loop for removing properties. Since `stateClone` is already a copy of the state, you can directly delete the keys from `stateClone`.